hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/TssMetrics.java

```diff
@@ -0,0 +1,104 @@
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * A class to track all the metrics related to TSS functionalities.
+ */
+@Singleton
+public class TssMetrics {
+    private static final String TSS_CANDIDATE_ROSTER_LIFECYCLE = "tss_candidate_roster_lifecycle";
+    private static final String TSS_CANDIDATE_ROSTER_LIFECYCLE_DESC = "the lifecycle of the current candidate roster";
+    private static final LongGauge.Config TSS_ROSTER_LIFECYCLE_CONFIG = new LongGauge.Config(
+                    "app", TSS_CANDIDATE_ROSTER_LIFECYCLE)
+            .withDescription(TSS_CANDIDATE_ROSTER_LIFECYCLE_DESC);
+    private final LongGauge tssCandidateRosterLifecycle;
+
+    private static final String TSS_MESSAGE_COUNTER_METRIC = "tss_message_total";
+    private static final String TSS_MESSAGE_COUNTER_METRIC_DESC = "total numbers of tss message transactions";
+    private static final Counter.Config TSS_MESSAGE_TX_COUNTER =
+            new Counter.Config("app", TSS_MESSAGE_COUNTER_METRIC).withDescription(TSS_MESSAGE_COUNTER_METRIC_DESC);
+    private final Counter tssMessageTxCounter;
+
+    private static final String TSS_VOTE_COUNTER_METRIC = "tss_vote_total";
+    private static final String TSS_VOTE_COUNTER_METRIC_DESC = "total numbers of tss vote transactions";
+    private static final Counter.Config TSS_VOTE_TX_COUNTER =
+            new Counter.Config("app", TSS_VOTE_COUNTER_METRIC).withDescription(TSS_VOTE_COUNTER_METRIC_DESC);
+    private final Counter tssVoteTxCounter;
+
+    /**
+     * Constructor for the TssMetrics
+     *
+     * @param metrics the {@link Metrics} object where all metrics will be registered
+     */
+    @Inject
+    public TssMetrics(@NonNull final Metrics metrics) {
+        requireNonNull(metrics, "metrics must not be null");
+
+        tssCandidateRosterLifecycle = metrics.getOrCreate(TSS_ROSTER_LIFECYCLE_CONFIG);
+        tssMessageTxCounter = metrics.getOrCreate(TSS_MESSAGE_TX_COUNTER);
+        tssVoteTxCounter = metrics.getOrCreate(TSS_VOTE_TX_COUNTER);
+    }
+
+    /**
+     * Increment counter metrics for TssVote or TssMessage transactions.
+     *
+     * @param functionality the TSS {@link HederaFunctionality} for which the metrics will be updated
+     */
+    public void updateTssTransactionMetrics(@NonNull final HederaFunctionality functionality) {
+        requireNonNull(functionality, "functionality must not be null");
+        if (functionality == HederaFunctionality.NONE) {
+            return;
+        }
+
+        if (functionality == TSS_MESSAGE) {
+            tssMessageTxCounter.increment();
+        } else if (functionality == TSS_VOTE) {
+            tssVoteTxCounter.increment();
+        }
+    }
+
+    /**
+     * Track when the candidate roster is set/
+     *
+     * @param lifecycle the time at which the candidate roster is set
+     */
+    public void updateCandidateRosterLifecycle(long lifecycle) {
```

Comment:
Nit: `final long lifecycle`

Is it worth checking that this value is > 0? (in this and the next method)